src/services/dashboard-service.js

```diff
@@ -0,0 +1,140 @@
+import by from 'thenby';
+import RoomStore from '../stores/room-store.js';
+import UserStore from '../stores/user-store.js';
+import LessonStore from '../stores/lesson-store.js';
+import DocumentStore from '../stores/document-store.js';
+import { FAVORITE_TYPE, USER_ACTIVITY_TYPE } from '../domain/constants.js';
+
+class DashboardService {
+  static get inject() { return [UserStore, DocumentStore, RoomStore, LessonStore]; }
+
+  constructor(userStore, documentStore, roomStore, lessonStore) {
+    this.userStore = userStore;
+    this.roomStore = roomStore;
+    this.lessonStore = lessonStore;
+    this.documentStore = documentStore;
+  }
+
+  async getUserActivities({ userId, limit = 0 }) {
+    const user = await this.userStore.getUserById(userId);
+
+    const createdRooms = await this.roomStore.getRoomsCreatedByUser(userId);
+    const joinedRooms = await this.roomStore.getRoomsJoinedByUser(userId);
+    const updatedRooms = await this.roomStore.getRoomsUpdatedByUser(userId);
+    const createdDocuments = await this.documentStore.getDocumentsMetadataByCreatedBy(userId);
```

Comment:
I was thinking a little bit about the naming pattern, and although `byCreatedBy` is the technically 100% correct name, it sounds unnatural. We could have a similar naming like in the room store with `getDocumentsMetadataCreatedByUser(userId)`, what do you think?

Reply:
Updated all such phrased functions, thanks!

Reply:
That's true, the `DocumentMetadata` is crooked anyway.